master/pkg/schemas/expconf/storage_config.go

```diff
@@ -131,6 +132,20 @@ type S3ConfigV0 struct {
 	RawAccessKey   *string `json:"access_key"`
 	RawSecretKey   *string `json:"secret_key"`
 	RawEndpointURL *string `json:"endpoint_url"`
+	RawPrefix      *string `json:"prefix"`
+}
+
+// Validate implements the check.Validatable interface.
```

Comment:
I don't think you need this.  I think catching it before launching experiments should be sufficient.

Where would this actually get called?

Reply:
Yeah, I couldn't understand these functions at first but after some digging (confirmed with Bradely), when we load up the master.yaml, we walk through all the structs and look for a `Validate()` method and call it. There is some recursive reflection voodoo here: `master/pkg/check/validate.go`, which handles all that.

We need this because if we don't have it and we put some invalid prefix in the `master.yaml`, we won't complain on startup, and will wait to fail until someone tries to launch an experiment ... which is misleading.